protocols/relay/examples/docker-compose.yml

```
@@ -0,0 +1,53 @@
+version: '3.4'
+services:
+  relay: 
+    image: libp2p-relay
+    environment:
+      - relay_command_arg_env=relay
+      - deterministic_peer_env=1
+      - address_env=/ip6/::/tcp/4444
+    build:
+      context: ../../../.
+      dockerfile: ./protocols/relay/examples/Dockerfile
+    networks:
+      - default
+
+  client-listen: 
+    image: libp2p-relay
+    environment:
+      - relay_command_arg_env=client-listen
+      - deterministic_peer_env=2
+      - address_env=/dns6/relay/tcp/4444/p2p/12D3KooWPjceQrSwdWXPyLLeABRXmuqt69Rg3sBYbU1Nft9HyQ6X/p2p-circuit
+    build:
+      context: ../../../.
+      dockerfile: ./protocols/relay/examples/Dockerfile
+    depends_on:
+        - "relay"
+    networks:
+      - default
+
+  client-dial: 
+    image: libp2p-relay
+    environment:
+      - relay_command_arg_env=client-dial
+      - deterministic_peer_env=3
+      - address_env=/dns6/relay/tcp/4444/p2p/12D3KooWPjceQrSwdWXPyLLeABRXmuqt69Rg3sBYbU1Nft9HyQ6X/p2p-circuit/p2p/12D3KooWH3uVF6wv47WnArKHk5p6cvgCJEb74UTmxztmQDc298L3
+    build:
+      context: ../../../.
+      dockerfile: ./protocols/relay/examples/Dockerfile
+    depends_on:
+      - "client-listen"
+    networks:
+      - default
+      
+networks:
+  default:
+    driver: bridge
+    enable_ipv6: true
+    ipam:
+      driver: default
+      config:
+        - subnet: 172.16.238.0/24
+          gateway: 172.16.238.1
+        - subnet: 2001:3984:3989::/64
+          gateway: 2001:3984:3989::1
```

```suggestion
  - network-a
  - network-b
```

Are these configuration options needed? In addition, as you suggested via mail, how about placing the `client-dial` and `client-listen` into different networks and the `relay` into both?

---

Without the ipam configuration the docker compose failed with
could not find an available, non-overlapping IPv6 address pool among the defaults to assign to the network.

---

How does this setup differ to the one below? Asked differently, why can the setup below work without these configuration options?

https://github.com/paritytech/polkadot/blame/1b82dc5b56e3970a0cfddaad17bbc7c487d5dbba/docker/sentry-docker-compose.yml#L101-L103

---

The difference is the ipv6 vs ipv4.
When using ipv6 an IPv6 subnet must be defined.
See:
https://medium.com/@skleeschulte/how-to-enable-ipv6-for-docker-containers-on-ubuntu-18-04-c68394a219a2
I suggest to separate between those 2 options using docker-compose. ipv4.yml and docker-compose. ipv6.yml

---

Tested this out locally. You are right. Sorry for the trouble! Two things:

- `enable_ipv6` does not seem to be available in docker-compose version `>=3.0` https://github.com/docker/docker.github.io/issues/3846. Unless I am missing something, would you mind downgrading the version to `2.1`?
    ```diff
    diff --git a/protocols/relay/examples/docker-compose.ipv6.yml b/protocols/relay/examples/docker-compose.ipv6.yml
    index 2d2b3c25..daf579bc 100644
    --- a/protocols/relay/examples/docker-compose.ipv6.yml
    +++ b/protocols/relay/examples/docker-compose.ipv6.yml
    @@ -1,7 +1,7 @@
    # This docker compose use the same example as the docker-compose.yml but with ipv6 instead of ipv4
    -version: '3.4'
    +version: '2.1'
    ```
- For the sake of simplicity, I would suggest removing the ipv4 docker-compose file.